New feature : count windows

Yinameah opened the pull request:
Added option to display a windows count in the tabline. This comes with some tweaking, i.e. :
- show only when multiple windows
- count only different buffers
- show delta - displays "filename.xxx [+2]" if there is 2 other windows on top of the filename.xxx one

Image for reference : 

![image](https://github.com/user-attachments/assets/3ef05675-35c2-4cc3-b55b-cc5b3e50eb7a)

Counting only different buffers, and "other" (meaning in the current tab there is "foo.py" plus two other different files in the same tab)

![image](https://github.com/user-attachments/assets/c030dfa8-ba05-40fe-a43b-780ce444ddce)

Yinameah committed:
New feature : count windows

Added option to display a windows count in the tabline.
This comes with some tweaking, i.e. :
- show only when multiple windows
- count only different buffers
- show delta - displays "filename.xxx [+2]" if there is 2 other windows
  on top of the filename.xxx one

seblyng commented on lua/tabline/utils.lua:
```diff
@@ -34,4 +34,67 @@ M.get_tabname = function(bufname, index)
     return vim.fn.fnamemodify(bufname, ':t')
 end
 
+M.is_val_in_table = function(val, tbl)
```
I think you can use: `vim.tbl_contains` instead of this?

Or in the case I see that it's used, I think `vim.list_contains` is a better fit

Yinameah replied:
Ah, good. Wasn't aware. Will change.

seblyng commented on lua/tabline/utils.lua:
```diff
@@ -34,4 +34,67 @@ M.get_tabname = function(bufname, index)
+    return false
+end
+
+M.is_focusable = function(win_id)
```
Make this a local function

Yinameah replied:
Makes perfect sense, will do

seblyng commented on lua/tabline/utils.lua:
```diff
@@ -34,4 +34,67 @@ M.get_tabname = function(bufname, index)
+    for _, win_id in pairs(win_list) do
+        if M.is_focusable(win_id) then
+            local bufnr = vim.api.nvim_win_get_buf(win_id)
+            local bt = vim.api.nvim_get_option_value('buftype', { buf = bufnr })
```
`vim.bo[bufnr].buftype`

Yinameah replied:
Yeah, I was trying to wrap my head around this api, but this is so much more readable indeed.

seblyng commented on lua/tabline/utils.lua:
```diff
@@ -34,4 +34,67 @@ M.get_tabname = function(bufname, index)
+        if opts.count_others then
+            -- if blacklisted buftype is currently focused, the [+x] doesn't make sense, so
+            -- we kind of count it anyway
+            local focused_bt = vim.api.nvim_get_option_value(
```
`vim.bo` here as well. And maybe create a local var bufnr before this so that it's not a super long line

seblyng reviewed (changes requested):
Could you also add a screenshot in the PR description just for reference?😅

Yinameah committed:
Rely on vim.list_contains instead of writing my own util function

Yinameah committed:
keep local utils functions local

Yinameah committed:
Use more readable neovim api

nvim_get_option_value is very length, and in that case can be replaced
by vim.bo[bufnr] which is much more readable.

Yinameah commented:
Ok, did the small changes requested. And I will also add the new options in the Readme

Yinameah committed:
Updated Readme

- added new options
- added lazy.nvim config sample

Yinameah commented:
Ok, done. Also added config sample for lazy.nvim as I use it.

seblyng commented on README.md:
````diff
@@ -29,6 +29,18 @@ Plug 'nvim-tree/nvim-web-devicons'             " Optional
+```lua
+return {
+    "seblj/nvim-tabline",
+    requires = { "nvim-tree/nvim-web-devicons" }, -- Optional
````
Isn't this `dependencies` and not`requires`? I think that is packer.nvim

Yinameah replied:
Yes, absolutely, don't know how I messed up this copy-paste.

seblyng reviewed (commented):
Code looks good now👍 haven't had the time to test this yet, but will do a quick test first
